test/e2e/commands/xmlupload/test_resources.py

```diff
@@ -2,63 +2,179 @@
-        assert len(class_with_everything_resource_graph) != 0
+    def test_class_with_everything_all_created(self, cls_with_everything_graph, class_with_everything_iri):
+        cls_iri = URIRef(class_with_everything_iri)
+        resource_iris = list(cls_with_everything_graph.subjects(RDF.type, cls_iri))
+        expected_number = 17
+        assert len(resource_iris) == expected_number
+
+    def test_resource_no_values_assert_triples_present(
+        self, cls_with_everything_graph, class_with_everything_iri, project_iri
+    ):
+        res_iri = _util_get_res_iri_from_label(cls_with_everything_graph, "resource_no_values")
+        number_of_triples = list(cls_with_everything_graph.triples((res_iri, None, None)))
+        assert next(cls_with_everything_graph.objects(res_iri, KNORA_API.attachedToProject)) == URIRef(project_iri)
+        assert next(cls_with_everything_graph.objects(res_iri, RDF.type)) == URIRef(class_with_everything_iri)
+        assert len(list(cls_with_everything_graph.objects(res_iri, KNORA_API.hasPermissions))) == 1
+        assert len(list(cls_with_everything_graph.objects(res_iri, KNORA_API.arkUrl))) == 1
+        assert len(list(cls_with_everything_graph.objects(res_iri, KNORA_API.versionArkUrl))) == 1
+        assert len(list(cls_with_everything_graph.objects(res_iri, KNORA_API.userHasPermission))) == 1
+        assert len(list(cls_with_everything_graph.objects(res_iri, KNORA_API.creationDate))) == 1
+        assert len(list(cls_with_everything_graph.objects(res_iri, KNORA_API.attachedToUser))) == 1
+        assert len(number_of_triples) == NUMBER_OF_RESOURCE_TRIPLES_WITHOUT_VALUES
+
+    def test_resource_no_permissions_specified(self, cls_with_everything_graph):
+        res_iri = _util_get_res_iri_from_label(cls_with_everything_graph, "resource_no_values")
+        actual_permissions = next(cls_with_everything_graph.objects(res_iri, KNORA_API.hasPermissions))
+        assert actual_permissions == DOAP_PERMISSIONS
+
+    def test_resource_with_open_permissions(self, cls_with_everything_graph):
+        res_iri = _util_get_res_iri_from_label(cls_with_everything_graph, "resource_no_values_open_permissions")
+        actual_permissions = next(cls_with_everything_graph.objects(res_iri, KNORA_API.hasPermissions))
+        assert actual_permissions == OPEN_PERMISSIONS
 
     @pytest.mark.usefixtures("_xmlupload")
-    def test_second_onto_class(self, second_onto_iri, auth_header, project_iri, creds):
+    def test_second_onto_class(self, second_onto_iri, onto_iri, auth_header, project_iri, creds):
         cls_iri_str = f"{second_onto_iri}SecondOntoClass"
-        resources = _util_request_resources_by_class(cls_iri_str, auth_header, project_iri, creds)
-        assert len(resources) != 0
+        g = _util_request_resources_by_class(cls_iri_str, auth_header, project_iri, creds)
+        resource_iris = list(g.subjects(RDF.type, URIRef(cls_iri_str)))
+        expected_number = 1
```

Comment:
In case that we change the XML one day, and add another resource of type `:SecondOntoClass`, the name `expected_number` won't help us much. I would choose a more telling name, e.g. `no_of_resources_of_type_SecondOntoClass`